src/com/google/enterprise/adaptor/DocumentHandler.java

```diff
@@ -774,9 +775,18 @@ public void respondNoContent() throws IOException{
       }
       if (state == State.NO_CONTENT) {
         startSending(false);
-      } else if (state == State.NO_CONTENT_TRANSFORMED_TO_NOT_FOUND) {
-        log.log(Level.INFO, "changed NO_CONTENT to NOT_FOUND {0}",
+      } else if (state == State.NO_CONTENT_TRANSFORMED_TO_NO_INDEX) {
+        log.log(Level.INFO, "changed NO_CONTENT to DO_NOT_INDEX {0}",
             docId.getUniqueId());
+        // Send no metadata, other than Robots noindex.
+        ex.getResponseHeaders().add("X-Robots-Tag", "noindex");
+        // Respond with 304 instead of 204 when talking with non GSA requests
+        // such as browsers.
+        HttpExchanges.respond(ex,
+            (requestIsFromFullyTrustedClient(ex)
+                ? HttpURLConnection.HTTP_NO_CONTENT
+                : HttpURLConnection.HTTP_NOT_MODIFIED),
+            "text/plain", (byte[]) null);
```

Comment:
This isn't a HEAD request, so perhaps we should follow the chunked examples elsewhere and use new byte[0] here instead of null. Since getOutputStream won't be called here, or it is an error if it is, I'm not sure whether we will get bitten by the double-close exception in HttpsExchanges.respond when the response is non-null, or not.